engine/model_test.go

```diff
@@ -175,6 +215,43 @@ func (s *BackendSuite) TestOCSPSettingsEq(c *C) {
 	}
 }
 
+func (s *BackendSuite) TestListenerSettingsEq(c *C) {
+	options := []struct {
+		a Listener
+		b Listener
+		e bool
+		c string
+	}{
+		{
+			a: Listener{},
+			b: Listener{},
+			e: true,
+			c: "empty",
+		},
+		{
+			a: Listener{Settings: &HTTPSListenerSettings{}},
+			b: Listener{Settings: &HTTPSListenerSettings{}},
+			e: true,
+			c: "emtpy https",
+		},
+		{
+			a: Listener{Settings: &HTTPSListenerSettings{TLS: TLSSettings{}}},
+			b: Listener{Settings: &HTTPSListenerSettings{TLS: TLSSettings{}}},
+			e: true,
+			c: "default https",
+		},
+		{
+			a: Listener{Settings: &HTTPSListenerSettings{TLS: TLSSettings{SessionTicketsDisabled: false}}},
+			b: Listener{Settings: &HTTPSListenerSettings{TLS: TLSSettings{}}},
+			e: true,
+			c: "different https",
+		},
+	}
+	for _, o := range options {
+		c.Assert((&o.a).SettingsEquals(&o.b), Equals, o.e, Commentf("TC: %v", o.c))
+	}
+}
```

Comment:
I would add some tests that expect `SettingsEquals` to return false. Otherwise

```go
func (l *Listener) SettingsEquals(o *Listener) bool {
        return true
}
```
would pass the test.